src/plugins/ui_core_views/dynamic_tables.ts

```diff
@@ -0,0 +1,254 @@
+import {
+  areZonesContinuous,
+  deepEquals,
+  getZoneArea,
+  isInside,
+  isZoneInside,
+  overlap,
+  toZone,
+  union,
+  unionPositionsToZone,
+  zoneToXc,
+} from "../../helpers";
+import { createFilter } from "../../helpers/table_helpers";
+import {
+  CellPosition,
+  Command,
+  CoreTable,
+  DynamicTable,
+  ExcelWorkbookData,
+  Filter,
+  FilterId,
+  Position,
+  Table,
+  TableId,
+  UID,
+  Zone,
+  invalidateEvaluationCommands,
+} from "../../types/index";
+
+import { UIPlugin } from "../ui_plugin";
+
+export class DynamicTablesPlugin extends UIPlugin {
+  static getters = [
+    "canCreateDynamicTableOnZones",
+    "doesZonesContainFilter",
+    "doesZonesContainTable",
+    "getFilter",
+    "getFilters",
+    "getTable",
+    "getTables",
+    "getTablesInZone",
+    "getTablesOverlappingZones",
+    "getFilterId",
+    "getFilterHeaders",
+    "isFilterHeader",
+  ] as const;
+
+  tables: Record<UID, Table[]> = {};
+
+  handle(cmd: Command) {
+    if (
+      invalidateEvaluationCommands.has(cmd.type) ||
+      (cmd.type === "UPDATE_CELL" && "content" in cmd) ||
+      cmd.type === "EVALUATE_CELLS"
+    ) {
+      this.tables = {};
+      return;
+    }
+    switch (cmd.type) {
+      case "CREATE_TABLE":
+      case "REMOVE_TABLE":
+      case "UPDATE_TABLE":
+      case "DELETE_CONTENT":
+        this.tables = {};
+        break;
+      case "DUPLICATE_SHEET":
+        this.tables[cmd.sheetIdTo] = this.computeTables(cmd.sheetIdTo);
+        break;
+    }
+  }
+
+  finalize() {
+    for (const sheetId of this.getters.getSheetIds()) {
+      if (!this.tables[sheetId]) {
+        this.tables[sheetId] = this.computeTables(sheetId);
+      }
+    }
+  }
+
+  private computeTables(sheetId: UID): Table[] {
+    const tables: Table[] = [];
+    const coreTables = this.getters.getCoreTables(sheetId);
+
+    // First we create the static tables, so we can use them to compute collision with dynamic tables
+    for (const table of coreTables) {
+      if (table.type === "dynamic") continue;
+      tables.push(this.coreTableToTable(sheetId, table));
```

Comment:
call to `coreTableToTable` not useful here
```suggestion
      tables.push(table);
```